physical/gcs/gcs_test.go

```diff
@@ -84,5 +84,183 @@ func TestGCSBackend(t *testing.T) {
 
 	physical.ExerciseBackend(t, b)
 	physical.ExerciseBackend_ListPrefix(t, b)
+}
+
+func TestGCSHABackend(t *testing.T) {
+	credentialsFile := os.Getenv("GOOGLE_APPLICATION_CREDENTIALS")
+
+	// projectID is only required for creating a bucket for this test
+	projectID := os.Getenv("GOOGLE_PROJECT_ID")
+
+	if credentialsFile == "" || projectID == "" {
+		t.SkipNow()
```

Comment:
We should probably fail the test or at least print a warning.

Reply:
Same for the rest of the backends, skipped if the required vars are not set